pkg/driver/driver_openstack.go

```diff
@@ -143,8 +146,50 @@ func (d *OpenStackDriver) Create() (string, string, error) {
 	var podNetworkIds = make(map[string]struct{})
 
 	if len(networkID) > 0 {
-		serverNetworks = append(serverNetworks, servers.Network{UUID: networkID})
+		// network port dealing if subnetID is specified
+
+		klog.V(3).Infof("existing network %s specified, need to pre-allocate ports ", networkID)
+
+		if subnetID != nil && len(*subnetID) > 0 {
+			// create port in given subnet
+			_, err := subnets.Get(nwClient, *subnetID).Extract()
```

Comment:
```go			
if _, err := subnets.Get(nwClient, *subnetID).Extract(); err != nil {
```

Reply:
Ok, done. I believe this is preferred because it limits the visibility of the err variable, right?